### PR TITLE
Switch to `clap` arg parser instead of `structopt` in utils

### LINKED ISSUE #2442: Use clap instead of structopt in utils
### File Location(s)

https://github.com/gear-tech/gear/blob/a2f6ac69cd73a911494f8a5a78f00e08c3d1a8f9/utils/node-loader/Cargo.toml#L31

--- 

- gear-node-loader
- runtime-fuzzer
- wasm-info

### Proposal

most of our CLI are using clap directly for now since it just support proc-macro as well last years, so using clap instead of structopt here as well for syncing dependencies

### PR DESCRIPTION
**Pull request by tltsutltsu**
Resolves #2442.

Changes made due to the transition:
- In `wasm-info`, the shorthand for `--hex` has been changed to `-H` from `-h`. This has been done because it conflicts with the shorthand for help in current version of clap. Another option is to disable both `--help` and `-h`, but this requires further discussion.

@breathx

### CODE REVIEW

**Review by shamilsan — APPROVED** [review]

File: utils/wasm-info/Cargo.toml [hunk 1]
```diff
@@ -4,6 +4,6 @@ version = "0.1.0"
 edition = "2021"
 
 [dependencies]
-hex.workspace = true
+hex = { workspace = true, features = ["std"]}
```
Comment (breathx) [inline utils/wasm-info/Cargo.toml:7]: why "std" feature now required?

**Review by breathx — APPROVED** [review]

File: utils/wasm-info/Cargo.toml (hunk 1 above)
Comment (tltsutltsu) [reply utils/wasm-info/Cargo.toml:7]: Dependencies for `wasm-info` were changed to be managed via workspace. Before this change, `wasm-info` relied on the `hex` crate with default features, but now it does not, which caused the crate to fail to build. To fix this, I added the minimal required feature set to allow `wasm-info` to work again

File: utils/node-loader/Cargo.toml [hunk 2]
```diff
@@ -28,9 +28,9 @@ parking_lot.workspace = true
 primitive-types = { workspace = true, features = ["scale-info"] }
 rand = { workspace = true, features = ["small_rng"] }
 reqwest.workspace = true
-structopt = "0.3.26" # TODO: use clap instead
 thiserror.workspace = true
 tokio = { workspace = true, features = [ "macros", "rt-multi-thread" ] }
 tracing.workspace = true
 tracing-appender.workspace = true
 tracing-subscriber = { workspace = true, features = [ "env-filter", "json" ] }
+clap = { workspace = true, features = ["derive"] }
```
[the remark this answers is not in the document]
Comment (tltsutltsu) [reply utils/node-loader/Cargo.toml]: Fixed and remembered

File: utils/runtime-fuzzer/src/bin/reproduce_fuzz.rs [hunk 3]
```diff
@@ -25,24 +25,24 @@
...
     path::PathBuf,
 };
-use structopt::StructOpt;
 
-#[derive(Debug, StructOpt)]
+#[derive(Debug, Parser)]
 pub struct Params {
     /// Path to the file, which contains seeds from previously run fuzzer.
-    #[structopt(short = "p", long, parse(from_os_str))]
+    #[arg(short = 'p', long)]
```
[the remark this answers is not in the document]
Comment (tltsutltsu) [reply utils/runtime-fuzzer/src/bin/reproduce_fuzz.rs]: Done

File: utils/wasm-info/src/main.rs [hunk 4]
```diff
@@ -1,20 +1,20 @@
...
     pub hex: bool,
 
     /// Show code size
-    #[structopt(long)]
+    #[arg(long)]
     pub size: bool,
 
     /// Strip custom sections
-    #[structopt(short, long)]
+    #[arg(short, long)]
```
[the remark this answers is not in the document]
Comment (tltsutltsu) [reply utils/wasm-info/src/main.rs]: Done

**Review by breathx — APPROVED** [review]

**Comment by breathx** [comment]
Merge actual master to pass the CI

**Comment by tltsutltsu** [comment]
Closed in favor of #2549

**Pull request closed without merging**